### PR TITLE
[K12] Picklist Updates to Case(Behavior Involvement) and Behavior Response Objects

### LINKED ISSUE #198: [K12] Picklist Updates to Case(Behavior Involvement) and Behavior Response Objects
**Note: FOR INTERNAL USE ONLY. For support questions, please use the EDA Chatter group in [The Power of Us Hub](https://powerofus.force.com).
For feature requests, please use the Ideas tab in [The Power of Us Hub](https://powerofus.force.com).**

* **Do you want to request a *feature* or report a *bug*?**
Feature 
* **What is the current behavior?**
As a school administrator, I cannot record response to an incident that involved one or more of my students either at school or not so that I can record and track behavior and the responses taken by my school. 

As a school administrator I cannot record where an incident occurred so that I can better track problem areas at my school.

* **What is the expected behavior?**
As a school administrator I can record the response to an incident that involved one or more of my students either at school or not so that I can record and track behavior and the responses taken by my school.

As a school administrator I can record where an incident occurred so that I can better track problem areas at my school.

* **What is the motivation / use case for changing the behavior?**
Data Integrity

### PR DESCRIPTION
**Pull request by jofsky**
# Critical Changes

# Changes
## New Behavior Response-Related Picklist Values
We've added new picklist values to the existing "Type" field on Behavior Response. 
- Detention
- Expulsion with Services
- In-School Suspension
- Out-of-School Suspension
- Removal from Classroom

We've also added these picklist values to the existing "Location" field on Case for Behavior Response. 
- Administrative offices area
- Athletic field or playground
- Auditorium
- Bus stop
- Cafeteria area
- Classroom
- Computer lab
- Hallway or stairs
-  Library/media center
- Locker room or gym areas
- Online
- Parking lot
- Restroom
- School bus
- Stadium
- Unknown
- Walking to or from school

For new installations, the new picklist values are automatically included. If you installed an earlier EDA release, manually add the new picklist values to the "Type" field on Behavior Response and to the "Location" field on Case for Behavior Response as indicated above. For information, check out [Add Custom Fields and Picklist Values](https://powerofus.force.com/s/article/EDA-Add-Custom-Picklist-Values).

# Issues Closed
Closes #198
# New Metadata
New picklist values for hed__Type__c on Behavior Response: 
- Detention
- Expulsion with Services
- In-School Suspension
- Out-of-School Suspension
- Removal from Classroom

New picklist values for hed__Location__c on Case: 
- Administrative offices area
- Athletic field or playground
- Auditorium
- Bus stop
- Cafeteria area
- Classroom
- Computer lab
- Hallway or stairs
-  Library/media center
- Locker room or gym areas
- Online
- Parking lot
- Restroom
- School bus
- Stadium
- Unknown
- Walking to or from school

# Deleted Metadata

# Testing Notes
Dev org: 
- Create a dev org and run dev_org flow and check that the new picklist values are showing up alphabetically and that "Other" is set at the end of the list. For Hed__Location__c, checking the picklist values via Incident record type, the "Other" is not set at the end of the list. 

Trial org: 
- Create a trial org and run trial_org flow and check that the new picklist values are showing up alphabetically and that "Other" is set at the end of the list. For Hed__Location__c, checking the picklist values via Incident record type, the "Other" is not set at the end of the list. 

- Create a trial org and run net_new_org flow and check that the new picklist values are showing up alphabetically and that "Other" is set at the end of the list. For Hed__Location__c, checking the picklist values via Incident record type, the "Other" is not set at the end of the list.

### CODE REVIEW

**Comment by kdy618** [comment]
@jofsky , do you know if we should test this against a trial org? I know we discovered some issues/inconsistency with a trial org on a different WI.

**Comment by kdy618** [comment]
@jstvz , could you kindly review the changes made in add_picklist_values.py and cumulus.yml file please? 

cc: @lifewithryan

**Comment by jofsky** [comment]
@jstvz can you review the namespace injection

**Comment by spinkelman** [comment]
@jstvz Based on @jofsky last comment, would you please review this PR and approve it so I can merge? 

QA passes based on the qa | qa_org, dev_namespaced | dev_org_namespaced, and trial | trial_org flows when accounting for the two new WI to address issues found.

Thanks!

**Comment by jofsky** [comment]
@spinkelman  I can't approve my own PR unfortunately, need you or @jstvz  to help me out.

File: tasks/add_picklist_values.py [hunk 1]
```diff
@@ -85,27 +86,43 @@ class AddPicklistValues(BaseSalesforceApiTask, Deploy):
...
             "required": False,
         },
-        "otherlast": {
-            "description": "If true, the Other value (if one exists) will remain at the end of the values",
+        "restricted": {
+            "description": "If picklist value is a required field",
+            "required": False,
+        },
+        "namespaced": {
+            "description": "If picklist value is a required field",
```
Comment (jstvz) [inline tasks/add_picklist_values.py:102]: Update this description.

File: tasks/add_picklist_values.py [hunk 2]
```diff
@@ -85,27 +86,43 @@ class AddPicklistValues(BaseSalesforceApiTask, Deploy):
...
+            "required": False,
+        },
         "sorted": {
             "description": "If true, sets the entire picklist to be sorted in alphabetical order",
             "required": False,
         },
-        "otherlast": {
-            "description": "If true, the Other value (if one exists) will remain at the end of the values",
+        "restricted": {
+            "description": "If picklist value is a required field",
```
Comment (jstvz) [inline tasks/add_picklist_values.py:98]: Update this description.

File: tasks/add_picklist_values.py [hunk 3]
```diff
@@ -239,57 +266,64 @@ def _build_package_xml(self, picklist_record_types, sobject, field):
...
-            if value.lower() in values_added:
-                self.logger.info(f"{value} is already a picklist value on {field}. Skipping over...")
-
-            else:
-                picklist_values_xml += picklist_value_template.format(
-                    name = escape(value), 
-                    default = False, 
-                    label = escape(value)
-                )
+            # if value.lower() in values_added:
```
Comment (jstvz) [inline tasks/add_picklist_values.py:318]: Remove dead code

**Comment by spinkelman** [comment]
QA Passes with inclusion of three additional WIs created for additional work.

Will close this WI

**Pull request merged**